core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceServiceUnitTestsSpec.scala

```diff
@@ -0,0 +1,307 @@
+        mock[BillingProfileManagerDAO],
+        mock[SamDAO],
+        mock[MultiCloudWorkspaceConfig],
+        mock[LeonardoDAO],
+        "MultiCloudWorkspaceService-test"
+      )(requestContext)
+    )
+
+    val billingProfile = mock[ProfileModel]
+    when(billingProfile.getCreatedDate).thenReturn(DateTime.now().toString)
+
+    val destWorkspace = mock[Workspace]
+    doReturn(Future(new Exception()))
+      .when(service)
+      .createNewWorkspaceRecord(
+        ArgumentMatchers.any(),
+        ArgumentMatchers.eq(destWorkspaceRequest),
+        ArgumentMatchers.eq(requestContext),
+        ArgumentMatchers.eq(WorkspaceState.Cloning)
+      )
+
+    intercept[Exception] {
+      Await.result(
+        service.cloneAzureWorkspaceAsync(sourceWorkspace, billingProfile, destWorkspaceRequest, requestContext),
+        Duration.Inf
+      )
+    }
+
+    verify(service, never()).deleteWorkspaceRecord(destWorkspace)
+  }
+
+  it should "create the async clone job from the result in WSM" in {
+    val sourceWorkspaceName = "source-name"
+    val sourceWorkspaceNamespace = "source-namespace"
+    val sourceWorkspace = Workspace.buildMcWorkspace(
+      sourceWorkspaceNamespace,
+      sourceWorkspaceName,
+      UUID.randomUUID().toString,
+      DateTime.now(),
+      DateTime.now(),
+      "creator",
+      Map(),
+      WorkspaceState.Ready
+    )
+    val destWorkspaceRequest = WorkspaceRequest("dest-namespace", "dest-name", Map())
+
+    val requestContext = mock[RawlsRequestContext]
+    when(requestContext.otelContext).thenReturn(None)
+    val userInfo = mock[UserInfo]
+    when(userInfo.userEmail).thenReturn(RawlsUserEmail("user-email"))
+    when(requestContext.userInfo).thenReturn(userInfo)
+    val workspaceManagerDAO = mock[WorkspaceManagerDAO]
+    val workspaceManagerResourceMonitorRecordDao = mock[WorkspaceManagerResourceMonitorRecordDao]
+
+    val service = spy(
+      new MultiCloudWorkspaceService(
+        requestContext,
+        workspaceManagerDAO,
+        mock[BillingProfileManagerDAO],
+        mock[SamDAO],
+        mock[MultiCloudWorkspaceConfig],
+        mock[LeonardoDAO],
+        mock[SlickDataSource],
+        "MultiCloudWorkspaceService-test",
+        workspaceManagerResourceMonitorRecordDao
+      )
+    )
+
+    val billingProfile = mock[ProfileModel]
+    when(billingProfile.getCreatedDate).thenReturn(DateTime.now().toString)
+
+    val destWorkspace = mock[Workspace]
+    doReturn(Future.successful(destWorkspace))
+      .when(service)
+      .createNewWorkspaceRecord(
+        ArgumentMatchers.any(),
+        ArgumentMatchers.eq(destWorkspaceRequest),
+        ArgumentMatchers.eq(requestContext),
+        ArgumentMatchers.eq(WorkspaceState.Cloning)
+      )
+
+    val wsmResult = new CloneWorkspaceResult().jobReport(new JobReport().id("test-id-that-isn't-a-uuid"))
+    when(workspaceManagerDAO.cloneWorkspace(any(), any(), any(), any(), any(), any(), any())).thenReturn(wsmResult)
```

Comment:
I think we should match on the expected arguments (and include policy information).